### PR TITLE
specify acl for S3 objects

### PR DESCRIPTION
**Pull request by ivan-aws**
Allows to specify the object ACL when using the S3 module. Uses `private` as the default, which was the implicit default before.

This allows to utilize different ACLs, like `bucket-owner-full-control` which is useful when operating with shared files.

Example:

```python
s3 = S3(<region>,<bucket>)

s3.put_object(
    s3_asset_key,
    file_asset_path,
    style="s3-url",
    pre_check=True,
    object_acl="bucket-owner-full-control",
)
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ivan-aws** [comment]
Updated tests in second commit. tests pass locally for me.

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/s3.py [hunk 1]
```diff
@@ -146,7 +154,8 @@ def _perform_put_object(self, key, file_path):
                 self.region,
             )
             with open(file_path, 'rb') as file_handler:
-                self.resource.Object(self.bucket, key).put(Body=file_handler)
+                self.resource.Object(self.bucket, key).put(ACL=object_acl,
+                                                            Body=file_handler)
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/s3.py:158]:
```suggestion
                self.resource.Object(self.bucket, key).put(
                    ACL=object_acl,
                    Body=file_handler,
                )
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/tests/test_s3.py [hunk 2]
```diff
@@ -301,7 +301,7 @@ def test_perform_put_object_success(logger, boto3_resource):
         )
         mock_file.assert_called_with(file_path, 'rb')
         s3_resource.Object.assert_called_once_with(s3_cls.bucket, object_key)
-        s3_object.put.assert_called_once_with(Body=mock_file.return_value)
+        s3_object.put.assert_called_once_with(ACL='private',Body=mock_file.return_value)
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/tests/test_s3.py:304]:
```suggestion
        s3_object.put.assert_called_once_with(ACL='private', Body=mock_file.return_value)
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/tests/test_s3.py [hunk 3]
```diff
@@ -393,7 +393,7 @@ def test_perform_put_object_failed(logger, boto3_resource):
 
         mock_file.assert_called_with(file_path, 'rb')
         s3_resource.Object.assert_called_once_with(s3_cls.bucket, object_key)
-        s3_object.put.assert_called_once_with(Body=mock_file.return_value)
+        s3_object.put.assert_called_once_with(ACL='private',Body=mock_file.return_value)
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/python/tests/test_s3.py:396]:
```suggestion
        s3_object.put.assert_called_once_with(ACL='private', Body=mock_file.return_value)
```

**Review by sbkok — CHANGES_REQUESTED** [review]
I've added a few style suggestions, nothing major.
Thanks for contributing this.

**Comment by ivan-aws** [comment]
Done!
As an alternative, may I recommend switching to an opinionated formatting tool, such as black? Formatting can then be automated as part of the PR process

**Comment by sbkok** [comment]
Actually, I am working on adding automated formatting.
However, since that changes almost every line in our repository, I am waiting for our current PRs to be merged first.
Such that we don't need to modify as many PRs.

So most likely around releasing 3.2.0 I will add that to the repository.

**Review by sbkok — APPROVED** [review]
LGTM!

**Review by sbkok — APPROVED** [review]
LGTM, thanks!

**Review by javydekoning — APPROVED** [review]
LGTM! Thanks.

**Pull request merged**